### PR TITLE
chore: increase builds to keep

### LINKED ISSUE #1449: Increase the number of retained builds
The project only keeps the last 20 builds, and we'd need to retain a bigger number: 100? Because the main pipeline is a used as a helper, used as downstream for multiple sources (Beats PR/merges, nightly builds), in one day we loose the entire history.

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR mimics what the APM Agent Ruby has in its JJBB configuration for the build.

## Why is it important?
We want to keep as many build executions as possible, as the branches in the main MBP are used as downstream by multiple projects: nightly builds, Beats, and the development of the project itself.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Closes #1449

### CODE REVIEW

**Comment by cachedout** [comment]
Given that each of these builds is generating ~45MB in artifacts, I think this should keep around 1.5GB of data around at any given time.

I'm sure that's probably fine but I think we should add @leathekd as a reviewer just to be sure that's fine from the CI side of things.

**Review by cachedout — APPROVED** [review]

File: .ci/jobs/e2e-testing-mbp.yml [hunk 1]
```diff
@@ -5,7 +5,11 @@
     description: Jenkins pipeline for the e2e-testing project
     view: E2E
     project-type: multibranch
-    days-to-keep: 21
+    logrotate:
+      daysToKeep: 30
+      numToKeep: 300
+    number-to-keep: 100
```
Comment (leathekd) [inline .ci/jobs/e2e-testing-mbp.yml:11]: Did you want these to be the same?

**Review by leathekd — APPROVED** [review]
Left a comment about the actual number to keep (100 vs 300).  Other than that, this LGTM!

File: .ci/jobs/e2e-testing-mbp.yml [hunk 2]
```diff
@@ -5,7 +5,11 @@
     description: Jenkins pipeline for the e2e-testing project
     view: E2E
     project-type: multibranch
-    days-to-keep: 21
+    logrotate:
+      daysToKeep: 30
+      numToKeep: 300
```
Comment (mdelapenya) [inline .ci/jobs/e2e-testing-mbp.yml:10]:
```suggestion
      numToKeep: 100
```

File: .ci/jobs/e2e-testing-mbp.yml (hunk 1 above)
Comment (mdelapenya) [reply .ci/jobs/e2e-testing-mbp.yml]: Let's started with 100 first, thanks for taking a look!

**Review by v1v — APPROVED** [review]
+1

qq: If we ship the logs to Elasticsearch then this won't be needed right? 

The envision with the Jenkins Opentelemetry will be to support this, but it's still an experimental feature -> https://opentelemetry.io/status/#logging

**Comment by mdelapenya** [comment]
> +1
> 
> qq: If we ship the logs to Elasticsearch then this won't be needed right?
> 
> The envision with the Jenkins Opentelemetry will be to support this, but it's still an experimental feature -> https://opentelemetry.io/status/#logging

At this moment we are not seeking only logs but test results and artifacts. Once everything is observed from the CI, and we have a nice UI to retrieve it, then it would be great to have them all in the Elasticsearch instance.

**Pull request merged**